src/Infrastructure/Model/Shopware6Media.php

```diff
@@ -37,6 +47,92 @@ public function setFileName(string $fileName): void
 
     public function jsonSerialize(): array
     {
-        return ['fileName' => $this->fileName];
+        $data = [];
+        if (!is_null($this->fileName)) {
+            $data['fileName'] = $this->fileName;
+        }
+
+        foreach ($this->translations as $translation) {
+            $data['translations'][$translation->getLanguageId()] = $translation->jsonSerialize();
+        }
+
+        return $data;
+    }
+
+    public function isModified(): bool
+    {
+        return $this->modified;
+    }
+
+    public function setModified(): void
+    {
+        $this->modified = true;
+    }
+
+    /**
+     * @param Shopware6MediaTranslation[] $translations
+     */
+    public function setTranslations(array $translations): void
+    {
+        $this->translations = $translations;
+    }
+
+    /**
+     * @return Shopware6MediaTranslation[]
+     */
+    public function getTranslations(): array
+    {
+        return $this->translations;
+    }
+
+    /**
+     * @param Shopware6MediaTranslation[] $translations
+     */
+    public function updateTranslated(array $translations): void
+    {
+        $processedLanguages = [];
+        foreach ($translations as $translation) {
+            $processedLanguages[] = $translation->getLanguageId();
+            if (array_key_exists($translation->getLanguageId(), $this->translations)) {
+                if ($this->isTranslationChanged($translation, $this->translations[$translation->getLanguageId()])) {
+                    $this->translations[$translation->getLanguageId()] = new Shopware6MediaTranslation(
+                        $this->translations[$translation->getLanguageId()]->getId(),
+                        $translation->getAlt() ?? null,
+                        $translation->getTitle() ?? null,
+                        $translation->getLanguageId()
+                    );
+                    $this->setModified();
+                }
+                continue;
+            }
+            $this->translations[$translation->getLanguageId()] = new Shopware6MediaTranslation(
+                null,
+                $translation->getAlt() ?? null,
+                $translation->getTitle() ?? null,
+                $translation->getLanguageId()
+            );
+            $this->setModified();
+        }
+        foreach ($this->translations as $language => $translation) {
+            if (!in_array($language, $processedLanguages) && !$this->isTranslationEmpty($translation)) {
+                $this->translations[$language] = new Shopware6MediaTranslation(
+                    $translation->getId(),
+                    null,
+                    null,
+                    $translation->getLanguageId()
+                );
+                $this->setModified();
+            }
+        }
```

Comment:
can't it be simplified a bit?

I mean we could implement early exit/continue, or omit checking if translation is changed, and probably more

Reply:
extracted one method to apply more visibility.
purpose of this method is to calculate if data are modified, do not think I can't simplify it more.